### PR TITLE
Migrate mlip codes to pyiron_contrib

### PR DESCRIPTION
**Pull request by jan-janssen**

### CODE REVIEW

File: pyiron_contrib/atomistic/mlip/cfgs.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
+# coding: utf-8
+# http://gitlab.skoltech.ru/shapeev/mlip-dev/blob/master/src/external/python/mlippy/cfgs.py
```
Comment (pmrv) [inline pyiron_contrib/atomistic/mlip/cfgs.py:2]: Are we allowed to redistribute this?

**Review by pmrv — CHANGES_REQUESTED** [review]

**Comment by jan-janssen** [comment]
```
/srv/conda/envs/notebook/lib/python3.7/site-packages/pyiron_contrib/atomistic/mlip/mlip.py in _write_test_set(self, file_name, cwd)
    320 
    321     def _write_test_set(self, file_name='testing.cfg', cwd=None):
--> 322         return self._write_configurations(file_name=file_name, cwd=cwd, respect_step=False, return_max_index=True)
    323 
    324     def _write_training_set(self, file_name='training.cfg', cwd=None):

/srv/conda/envs/notebook/lib/python3.7/site-packages/pyiron_contrib/atomistic/mlip/mlip.py in _write_configurations(self, file_name, cwd, respect_step, return_max_index)
    240                     track_lst.append(name)
    241                 continue
--> 242             original_dict = {el: ind for ind, el in enumerate(sorted(ham['input/structure/species']))}
    243             species_dict = {ind: original_dict[el] for ind, el in enumerate(ham['input/structure/species'])}
```

**Comment by pmrv** [comment]
> ```
> /srv/conda/envs/notebook/lib/python3.7/site-packages/pyiron_contrib/atomistic/mlip/mlip.py in _write_test_set(self, file_name, cwd)
>     320 
>     321     def _write_test_set(self, file_name='testing.cfg', cwd=None):
> --> 322         return self._write_configurations(file_name=file_name, cwd=cwd, respect_step=False, return_max_index=True)
>     323 
>     324     def _write_training_set(self, file_name='training.cfg', cwd=None):
> 
> /srv/conda/envs/notebook/lib/python3.7/site-packages/pyiron_contrib/atomistic/mlip/mlip.py in _write_configurations(self, file_name, cwd, respect_step, return_max_index)
>     240                     track_lst.append(name)
>     241                 continue
> --> 242             original_dict = {el: ind for ind, el in enumerate(sorted(ham['input/structure/species']))}
>     243             species_dict = {ind: original_dict[el] for ind, el in enumerate(ham['input/structure/species'])}
> ```

Ah, I forgot to mention that the training database needs to be regenerated, because I previously didn't set `TrainingContainer.__name__`, so jobs loaded from the old project end up having "GenericJob" set.  You should be able to regenerate them with

```python
new.include_dataset(old.to_pandas())
```

**Comment by jan-janssen** [comment]
Yes, That is what I also noticed https://github.com/jan-janssen/pyiron-mlip/commit/033e5758c4c94a403b6f5dc3a8f6fa891690a5b9 now the only limitation are the untrained potentials as the legacy version complains about the ones we have in the MPIE resources

**Comment by pmrv** [comment]
It's working now and fitting is quick.  I haven't tested the fitting quality though.  No changes to the notebook are necessary.

I force pushed over your commit, because if I understand it correctly that is already fixed by the updated project.  Feel free to rebase it on top if that's not the case.

**Comment by jan-janssen** [comment]
For me it is now working - here is an example how to do the fitting https://github.com/jan-janssen/pyiron-mlip

**Comment by jan-janssen** [comment]
@pmrv Can you take another look? I would like to merge this soon.

File: pyiron_contrib/atomistic/mlip/mlip.py [hunk 2]
```diff
@@ -388,10 +397,10 @@ def load_default(self, file_content=None):
         if file_content is None:
             file_content = '''\
 $MLP_COMMAND_PARALLEL train --energy-weight=energy_auto --force-weight=force_auto --stress-weight=stress_auto --max-iter=iteration_auto start.mtp training.cfg > training.log
-$MLP_COMMAND_SERIAL calc-grade Trained.mtp_ training.cfg testing.cfg grades.cfg --mvs-filename=state.mvs > grading.log
-$MLP_COMMAND_SERIAL select-add Trained.mtp_ training.cfg testing.cfg diff.cfg > select.log
-cp training.cfg training_new.cfg 
-cat diff.cfg >> training_new.cfg
+# $MLP_COMMAND_SERIAL calc-grade Trained.mtp_ training.cfg testing.cfg grades.cfg --mvs-filename=state.mvs > grading.log
```
Comment (pmrv) [inline pyiron_contrib/atomistic/mlip/mlip.py]: Are these not available in the legacy version?

Comment (jan-janssen) [reply pyiron_contrib/atomistic/mlip/mlip.py]: No - but I made it optional in the latest commit. They have some calc-error function but the syntax is different, so for the scope of the workshop I would suggest to validate the fit manually, as we anyway can not compare the residual computed by different codes.

**Review by pmrv — APPROVED** [review]
Looks good to me otherwise.

**Pull request merged**